src/pages/Overview/OverviewToolbar.tsx

```diff
@@ -83,14 +83,14 @@ export class OverviewToolbar extends React.Component<Props, State> {
     };
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: Props) {
     // ensure redux state and URL are aligned
     HistoryManager.setParam(URLParam.DURATION, String(this.props.duration));
     HistoryManager.setParam(URLParam.POLL_INTERVAL, String(this.props.refreshInterval));
 
     const urlSortField = ListPagesHelper.currentSortField(Sorts.sortFields);
     const urlIsSortAscending = ListPagesHelper.isCurrentSortAscending();
-    if (!this.paramsAreSynced(urlSortField, urlIsSortAscending)) {
+    if (!this.paramsAreSynced(urlSortField, urlIsSortAscending) || this.props.duration !== prevProps.duration) {
```

Comment:
Is this to replace the fact that onRefresh is no longer called when a duration is selected in the dropdown?

Reply:
Yes, instead of connecting that callback, we "listen" for changes on `duration`